File: src/main/frontend/src/app/models/dto/request/questionRequest.ts
import {QuestionType} from "../../enums/questionType";
import {AnswerRequest} from "./answerRequest";

export interface QuestionRequest {
  questionText: string,
  questionType: QuestionType,
  order: number,
  inputMinLimit: number | null,
  inputMaxLimit: number | null,
  answers: AnswerRequest[]
}

File: src/main/backend/src/main/java/io/server/backend/dto/request/QuestionRequest.java
package io.server.backend.dto.request;

import io.server.backend.model.enums.QuestionType;

import java.util.List;

public record QuestionRequest(
        String questionText,
        QuestionType questionType,
        Integer order,
        Integer inputMinLimit,
        Integer inputMaxLimit,
        List<AnswerRequest> answers
) { }
